BRMS-34 adding metrics class and unit tests

adamconder opened the pull request:
- Adding metrics class and methods to time and status count connector.

adamconder committed:
BRMS-34 adding metrics class and unit tests

chris-ianson commented on app/uk/gov/hmrc/brm/metrics/Metrics.scala:
```diff
@@ -0,0 +1,58 @@
+  Logger.info("[Metrics][constructor] Preloading metrics keys")
+
+  // register metrics with timer and counter
+  Seq(
```
@adamconder @manishsw 

Do we need to think about the key format/names? Do the keys below relate to the proxy response or the birth-registration-matching response to the consuming service?

adamconder replied:
@chris-ianson The keys below we're the responses from the proxy. We can add additional keys if necessary for the matching responses.

adamconder committed:
BRMS-34: adding metrics utility and implementation

adding metrics to:

- BirthEventController
- LookupService
- Connectors (response times, status codes)

adamconder committed:
BRMS-34 adding dependencies

adamconder commented:
@chris-ianson I've updated the PR with additional test cases and now works across all connectors.

chris-ianson commented on project/MicroServiceBuild.scala:
```diff
@@ -27,10 +27,10 @@ private object AppDependencies {
   private val mockito = "1.9.5"
-  private val metricsPlayVersion = "0.2.1"
+  private val metricsPlayVersion = "2.3.0_0.2.1"
+  private val metricsGraphite = "3.0.2"
```
Remove metricsGraphite

chris-ianson commented on project/MicroServiceBuild.scala:
```diff
@@ -39,7 +39,8 @@ private object AppDependencies {
     "uk.gov.hmrc" %% "domain" % domainVersion,
-    "com.kenshoo" %% "metrics-play" % metricsPlayVersion
+    "com.kenshoo" %% "metrics-play" % metricsPlayVersion,
+    "com.codahale.metrics" % "metrics-graphite" % metricsGraphite
```
Remove graphite

chris-ianson commented on test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala:
```diff
@@ -0,0 +1,224 @@
+
+    "have a 404 status count for proxy" in {
+      val metrics = GRONIMetrics
+      metrics.connectorStatus(404)
```
Ah, so we are a little more specific with error codes. Do you want us to log errors outside of 200, 400 and 500 (if applicable)

adamconder replied:
@chris-ianson It would be useful if you log exceptions outside of 400 etc.

adamconder replied:
Updated these :)

chris-ianson reviewed (changes requested):
Looks great, just a couple of comments

adamconder committed:
BRMS-34 removing dependency

removing dependency from build.sbt
